schemas/vault-config/instance-1.yml

```diff
@@ -0,0 +1,54 @@
+---
+"$schema": /metaschema-1.json
+version: '1.0'
+
+type: object
+additionalProperties: false
+properties:
+  "$schema":
+    type: string
+    enum:
+    - /vault-config/instance-1.yml
+  labels:
+    "$ref": "/common-1.json#/definitions/labels"
+  name:
+    type: string
+  description:
+    type: string
+  authType:
+    type: string
+  address:
+    type: string
+  roleID:
+    "$ref": "/common-1.json#/definitions/vaultSecret"
+  secretID:
+    "$ref": "/common-1.json#/definitions/vaultSecret"
+  token:
+    "$ref": "/common-1.json#/definitions/vaultSecret"
```

Comment:
since all these will likely be in the same secret, do we want to define a single field?

Reply:
I agree that it is redundant to specify secret path multiple times across different `VaultSecrets`, since it is highly likely the fields will be within the same secret. However, I believe explicitly defining the `field` attribute within each `VaultSecret` is less error prone/management.

The `ReadSecret` function pulls the full json for a particular secret path and `field` is utilized to quickly reference a specific value. Without defining fields, we would need to enforce naming conventions for each attribute when originally creating the vault secret (ex: role id should be entered as `role-id` etc.) and vault-manager logic would need to be hard-coded to specifically look for `role-id` or whatever we define.

By allowing users to define the name to map to roleID etc., we don't need to bother with enforcing naming. `roleID` can map to whatever field name they defined. This is demonstrated in the refactored test suite in which one instance defines token field as name `token` and the other defines name to look for as `rootToken`.